core/src/navigation/ContextSwitcher.html

```diff
@@ -0,0 +1,194 @@
+  import { getNegatedBoolString, trimLeadingSlash } from '../utilities/helpers.js';
+  import * as Routing from '../services/routing.js';
+
+  export default {
+    async oncreate() {
+      const contextSwitcherConfig = getConfigValue('navigation.contextSwitcher');
+      this.set({
+        contextSwitcherEnabled: !!contextSwitcherConfig,
+        config: contextSwitcherConfig
+      });
+
+      if (contextSwitcherConfig) {
+        const actions = await getConfigValueAsync('navigation.contextSwitcher.actions');
+        this.set({
+          actions,
+          fallbackLabelResolver: contextSwitcherConfig.fallbackLabelResolver
+        });
+        // options are loaded lazy by default
+        if (contextSwitcherConfig.lazyloadOptions === false) {
+          await this.fetchOptions();
+        }
+      }
+
+      await this.setSelectedContext(Routing.getCurrentPath());
+      Routing.addRouteChangeListener(() => this.setSelectedContext(Routing.getCurrentPath()));
+    },
+    data() {
+      return {
+        dropDownStates: {},
+        options: []
+      };
+    },
+    computed: {
+      dropDownStatesNegated({ dropDownStates }) {
+        const negated = {};
+        for (const key in dropDownStates) {
+          if (dropDownStates.hasOwnProperty(key)) {
+            negated[key] = getNegatedBoolString(dropDownStates[key]);
+          }
+        }
+        return negated;
+      }
+    },
+    helpers: {
+      getMatchingNodeName: (options, config, id) => {
+        return ContextSwitcherHelpers.getMatchingNodeName(options, config, id);
+      }
+    },
+    methods: {
+      async fetchOptions() {
+        let options = this.get().options;
+        if (!options.length || getConfigValue('navigation.contextSwitcher.lazyloadOptions') !== false) {
+          const contextSwitcherOptions = await getConfigValueAsync('navigation.contextSwitcher.options');
+          options = await ContextSwitcherHelpers.generateSwitcherNav(this.get().config, contextSwitcherOptions);
+          this.set({options});
+        }
+      },
+      async setSelectedContext(currentPath) {
+        let selectedLabel;
+        const parentNodePath = this.get().config.parentNodePath;
+        const inContextBasePath = parentNodePath && currentPath.indexOf(parentNodePath) !== -1;
+        if (inContextBasePath) {
+          // we are inside the context switcher base path
+          const truncatedPath = trimLeadingSlash(currentPath.replace(parentNodePath, ''));
+          const selectedId = truncatedPath.split('/')[0];
+          selectedLabel = await ContextSwitcherHelpers.getMatchingNodeName(this.get().options, this.get().config.fallbackLabelResolver, selectedId);
+        }
+        this.set({
+          selectedLabel
+        })
+      },
+      async onActionClick(node) {
+        if (node.clickHandler) {
+          const result = await getConfigValueFromObjectAsync({result: node.clickHandler(node)}, 'result');
```

Comment:
Can´t this just be `const result = await node.clickHandler()`? Apart from that, I would remove `node` input parameter, since it is not used in the extended configuration.